framework/infrastructure/traffic_director.py

```diff
@@ -679,32 +724,60 @@ def find_unused_forwarding_rule_port(
         # TODO(sergiitk): custom exception
         raise RuntimeError("Couldn't find unused forwarding rule port")
 
-    def create_forwarding_rule(self, src_port: int):
-        name = self.make_resource_name(self.FORWARDING_RULE_NAME)
+    def create_forwarding_rule(
+        self, src_port: int, ipv6=False, ip_address="0.0.0.0"
+    ):
+        if ipv6:
+            name = self.make_resource_name(self.FORWARDING_RULE_NAME_IPV6)
+            target_proxy = self.target_proxy_v6
+        else:
+            name = self.make_resource_name(self.FORWARDING_RULE_NAME)
+            target_proxy = self.target_proxy
+
         src_port = int(src_port)
         logging.info(
-            'Creating forwarding rule "%s" in network "%s": 0.0.0.0:%s -> %s',
+            'Creating forwarding rule "%s" in network "%s": %s:%s -> %s',
             name,
             self.network,
             src_port,
-            self.target_proxy.url,
+            ip_address,
```

Comment:
`src_port` should be put after `ip_address`.

Reply:
Should be fixed after refactor